funccount t:sched:sched_fork cause invalid argument

I run `python tools/funccount.py t:sched:sched_fork` , get the error message :

python tools/funccount.py t:sched:sched_fork
/virtual/main.c:3:1: error: could not open bpf map: Invalid argument
is maps/array map type enabled in your kernel?
BPF_TABLE("array", int, u64, counts, 0);
^
/virtual/include/bcc/helpers.h:53:4: note: expanded from macro 'BPF_TABLE'
}; \
   ^
1 error generated.
Failed to compile BPF module

my system is  Ubuntu 16.4 LTS and  kernel is latest kernel :

Linux ojl 4.9.0-040900rc2-generic #201610232131 SMP Mon Oct 24 01:32:49 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux

--- a/tools/funccount.py
+++ b/tools/funccount.py
@@ -107,10 +107,6 @@ class Probe(object):
         elif self.type == "u":
             pass    # Nothing to do -- attach already happened in `load`
 
-        if self.matched == 0:
-            raise Exception("No functions matched by pattern %s" %
-                            self.pattern)
-
     def _add_function(self, template, probe_name):
         new_func = "trace_count_%d" % self.matched
         text = template.replace("PROBE_FUNCTION", new_func)
@@ -198,6 +194,10 @@ BPF_TABLE("array", int, u64, counts, NUMLOCATIONS);
                                     str(len(self.trace_functions)))
         if debug:
             print(bpf_text)
+
+        if self.matched == 0:
+            raise Exception("No functions matched by pattern %s" %
+                            self.pattern)
 
         self.bpf = BPF(text=bpf_text,
                        usdt_contexts=[self.usdt] if self.usdt else [])

